### PR TITLE
US120707: hook up settings props and save to LMS

### PR DESCRIPTION
**Pull request by johngwilkinson**
Quality Notes
- functional
  - visit settings page, hit save, and confirm that the expected request is sent (nothing else is hooked up yet)
- security: n/a
- devops: n/a
- perf: n/a
- ux/docs: n/a

### CODE REVIEW

**Comment by johngwilkinson** [comment]
FYI @MykolaGalian @NicholasHallman @rohitvinnakota @Vitalii-Misechko

**Review by anhill-D2L — APPROVED** [review]

**Review by bemailloux — APPROVED** [review]
I had reviewed this yesterday but forgot to submit :/

File: engagement-dashboard.js [hunk 1]
```diff
@@ -443,13 +457,17 @@ class EngagementDashboard extends Localizer(MobxLitElement) {
...
-				includeRoles: this.includeRoles.split(',').filter(x => x).map(Number)
+				includeRoles: this._parsedIncludeRoles
 			});
 		}
 
 		return this.__serverData;
 	}
 
+	get _parsedIncludeRoles() {
+		return this.includeRoles.split(',').filter(x => x).map(Number);
```
[the remark this answers is not in the document]
Comment (johngwilkinson) [reply engagement-dashboard.js:468]: That's right. Specifically, `''.split(,)` is `[null]`, so the `filter` handles that corner case.

File: model/lms.js [hunk 2]
```diff
@@ -125,3 +126,39 @@ export function fetchLastSearch(selectedSemesterIds) {
...
+	});
+
+	const url = new URL(saveSettingsEndpoint, window.location.origin);
+	await fetch(url.toString(), {
+		method: 'PUT',
+		headers: {
+			'Content-Type': 'application/json'
+		},
+		body: JSON.stringify(settings)
+	});
```
[the remark this answers is not in the document]
Comment (johngwilkinson) [reply model/lms.js:163]: It should be the impersonated user, because that's what the injected context object will give us. I haven't tested, though.

**Pull request merged**